code/datums/world_topic.dm

```diff
@@ -0,0 +1,167 @@
+
+// TOPICS
+
+/datum/world_topic/ping
+	keyword = "ping"
+	log = FALSE
+
+/datum/world_topic/ping/Run(list/input)
+	. = 0
+	for (var/client/C in GLOB.clients)
+		++.
+
+/datum/world_topic/playing
+	keyword = "playing"
+	log = FALSE
+
+/datum/world_topic/playing/Run(list/input)
+	return GLOB.player_list.len
+
+/datum/world_topic/pr_announce
+	keyword = "announce"
+	require_comms_key = TRUE
+	var/static/list/PRcounts = list()	//PR id -> number of times announced this round
+
+/datum/world_topic/pr_announce/Run(list/input)
+	var/list/payload = json_decode(input["payload"])
+	var/id = "[payload["pull_request"]["id"]]"
+	if(!PRcounts[id])
+		PRcounts[id] = 1
+	else
+		++PRcounts[id]
+		if(PRcounts[id] > PR_ANNOUNCEMENTS_PER_ROUND)
+			return
+
+	var/final_composed = "<span class='announce'>PR: [input[keyword]]</span>"
+	for(var/client/C in GLOB.clients)
+		C.AnnouncePR(final_composed)
+
+/datum/world_topic/ahelp_relay
+	keyword = "ahelp_relay"
+	require_comms_key = TRUE
+
+/datum/world_topic/ahelp_relay/Run(list/input)
+	relay_msg_admins("<span class='adminnotice'><b><font color=red>HELP: </font> [input["source"]] [input["message_sender"]]: [input["message"]]</b></span>")
+
+/datum/world_topic/comms_console
+	keyword = "comms_console"
+	require_comms_key = TRUE
+
+/datum/world_topic/comms_console/Run(list/input)
+	minor_announce(input["message"], "Incoming message from [input["message_sender"]]")
+	for(var/obj/machinery/computer/communications/CM in GLOB.machines)
+		CM.overrideCooldown()
+
+/datum/world_topic/news_report
+	keyword = "news_report"
+	require_comms_key = TRUE
+
+/datum/world_topic/news_report/Run(list/input)
+	minor_announce(input["message"], "Breaking Update From [input["message_sender"]]")
+
+/datum/world_topic/server_hop
+	keyword = "server_hop"
+
+/datum/world_topic/server_hop/Run(list/input)
+	var/expected_key = input[keyword]
+	for(var/mob/dead/observer/O in GLOB.player_list)
+		if(O.key == expected_key)
+			if(O.client)
+				new /obj/screen/splash(O.client, TRUE)
+			break
+
+/datum/world_topic/adminmsg
+	keyword = "adminmsg"
+
+/datum/world_topic/adminmsg/Run(list/input)
+	return IrcPm(input[keyword], input["msg"], input["sender"])
+
+/datum/world_topic/namecheck
+	keyword = "namecheck"
+
+/datum/world_topic/namecheck/Run(list/input)
+	var/datum/server_tools_command/namecheck/NC = new
+	return NC.Run(input["sender"], input["namecheck"])
+
+/datum/world_topic/adminwho
+	keyword = "namecheck"
+
+/datum/world_topic/adminwho/Run(list/input)
+	return ircadminwho()
+
+/datum/world_topic/status
+	keyword = "status"
+
+/datum/world_topic/status/Run(list/input)
+	. = list()
+	.["version"] = GLOB.game_version
+	.["mode"] = GLOB.master_mode
+	.["respawn"] = config ? !CONFIG_GET(flag/norespawn) : FALSE
+	.["enter"] = GLOB.enter_allowed
+	.["vote"] = CONFIG_GET(flag/allow_vote_mode)
+	.["ai"] = CONFIG_GET(flag/allow_ai)
+	.["host"] = world.host ? world.host : null
+	.["active_players"] = get_active_player_count()

```

Comment:
remove this while you're in here. active_players turned out to be an exploitable info leak and was removed from the lobby tab and the game banners for that reason.